src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -29,18 +65,50 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
                 return false;
             }
 
-            var formattingService = document.GetLanguageService<IEditorFormattingService>();
-            if (formattingService == null || !formattingService.SupportsFormatDocument)
-            {
-                return false;
-            }
-
             using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Formatting_document))
             {
-                Format(args.TextView, document, null, context.OperationContext.UserCancellationToken);
+                var cancellationToken = context.OperationContext.UserCancellationToken;
+
+                var docOptions = document.GetOptionsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+
+                using (var transaction = new CaretPreservingEditTransaction(
+                    EditorFeaturesResources.Formatting, args.TextView, _undoHistoryRegistry, _editorOperationsFactoryService))
+                {
+                    var codeCleanupService = document.GetLanguageService<ICodeCleanupService>();
+
+                    if (codeCleanupService == null)
+                    {
+                        Format(args.TextView, document, selectionOpt: null, cancellationToken);
+                    }
+                    else if (!docOptions.GetOption(CodeCleanupOptions.AreCodeCleanupRulesConfigured))
```

Comment:
ah. you already have the option. then you probably doesn't need the field?

Reply:
hmm.. this might be different option? anyway, I think you should have 1 option that persisted between VS sessions, that make sure 1 user only see the info 1 time ever.